core/src/main/java/org/jruby/ext/socket/RubyIPSocket.java

```diff
@@ -46,51 +48,37 @@
  * @author <a href="mailto:ola.bini@ki.se">Ola Bini</a>
  */
 @JRubyClass(name="IPSocket", parent="BasicSocket")
-public class RubyIPSocket extends RubyBasicSocket {
+public abstract class RubyIPSocket extends RubyBasicSocket {
```

Comment:
`IPSocket` is still `allocate`-able ... not that it matters much (probably doesn't at all)
```
2.3.1 :029 > IPSocket.new
NoMethodError: undefined method `initialize' for #<IPSocket:0x0000000128ed70>
	from (irb):29:in `new'
	from (irb):29
	from /opt/local/rvm/rubies/ruby-2.3.1/bin/irb:11:in `<main>'
2.3.1 :030 > IPSocket.allocate
 => #<IPSocket:0x000000012574b0> 
```